Client/Application/LeanplumIntegration.swift

```diff
@@ -0,0 +1,201 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import AdSupport
+import Shared
+import Leanplum
+
+private let LeanplumEnvironmentKey = "LeanplumEnvironment"
+private let LeanplumAppIdKey = "LeanplumAppId"
+private let LeanplumKeyKey = "LeanplumKey"
+
+private let log = Logger.browserLogger
+
+private enum LeanplumEnvironment: String {
+    case development = "development"
+    case production = "production"
+}
+
+enum LeanplumEventName: String {
+    case firstRun = "First Run"
+    case secondRun = "Second Run"
+    case openedApp = "Opened App"
+    case openedLogins = "Opened Login Manager"
+    case openedBookmark = "Opened Bookmark"
+    case openedNewTab = "Opened New Tab"
+    case interactWithURLBar = "Interact With Search URL Area"
+    case savedBookmark = "Saved Bookmark"
+    case openedTelephoneLink = "Opened Telephone Link"
+    case openedMailtoLink = "Opened Mailto Link"
+    case downloadedImage = "Download Media - Saved Image"
+    case savedLoginAndPassword = "Saved Login and Password"
+}
+
+private enum SupportedLocales: String {
+    case US = "en_US"
+    case DE = "de"
+    case UK = "en_GB"
+    case CA_EN = "en_CA"
+    case AU = "en_AU"
+    case TW = "zh_TW"
+    case HK = "en_HK"
+    case SG_EN = "en_SG"
```

Comment:
@bkmunar will fix this up later - no rush though: https://bugzilla.mozilla.org/show_bug.cgi?id=1367963